src/core/qgsmaplayer.cpp

```diff
@@ -151,7 +151,11 @@ QgsMapLayer::LayerFlags QgsMapLayer::flags() const
 
 void QgsMapLayer::setFlags( QgsMapLayer::LayerFlags flags )
 {
+  if ( flags == mFlags )
+    return;
+
   mFlags = flags;
+  emit flagsChanged();
```

Comment:
flags are also changed in readLayerStyle at line 418 and 420.
I was a bit lazy....let's call setFlags.

Reply:
Addressed in 44611d7